packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1074,6 +1054,42 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
   private isValueEmpty = (value: ShorthandValue | ShorthandCollection) => {
     return _.isArray(value) ? value.length < 1 : !value
   }
+
+  private getHighlightedIndexOnArrowKeyOpen = (
+    changes: StateChangeOptions<ShorthandValue>,
+  ): number => {
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowUp) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      return _.isNumber(this.state.highlightedIndex)
+        ? (() => {
+            const newIndex = this.state.highlightedIndex - 1
+            return newIndex < 0 ? itemsLength - 1 : newIndex
+          })()
+        : itemsLength - 1
+    }
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowDown) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      return _.isNumber(this.state.highlightedIndex)
+        ? (() => {
+            const newIndex = this.state.highlightedIndex + 1
+            return newIndex >= itemsLength ? 0 : newIndex
+          })()
+        : 0
+    }
+    return undefined
```

Comment:
This method returns `undefined` while `getHighlightedIndexOnClose()` returns `null` 💭

Reply:
May be in this case we should return the current value?

Reply:
it's like this because when the menu opens by arrow up/down, we want to modify the highlighted index already saved in state. otherwise, it needs to be left alone, so it will be opened at whatever it was set in the close function.

when the menu closes, we want the index to be either selectedItem for single selection, 0 in case boolean prop is passed with true, or null if nothing else, so we make sure the dropdown opens with no highlight.